include/alpaka/rand/RandGenericSycl.hpp

```diff
@@ -0,0 +1,238 @@
+/* Copyright 2022
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#if defined(ALPAKA_ACC_SYCL_ENABLED)
+
+#    include <alpaka/core/BoostPredef.hpp>
+#    include <alpaka/core/Concepts.hpp>
+#    include <alpaka/dev/DevGenericSycl.hpp>
+#    include <alpaka/rand/Traits.hpp>
+
+// Backend specific imports.
+#    if defined(ALPAKA_ACC_SYCL_ENABLED)
+#        include <CL/sycl.hpp>
+#        include <oneapi/mkl/rng.hpp>
+
+#        include <oneapi/dpl/random>
+#    endif
```

Comment:
```suggestion
#        include <CL/sycl.hpp>
#        include <oneapi/mkl/rng.hpp>

#        include <oneapi/dpl/random>
```
Macro already checked above.

Reply:
Should we put this in another macro not to force the user to have `mkl` and `dpl`? If one is using `clang` maybe doesn't have those libraries.

Reply:
I leave this to @j-stephan

Reply:
In general, I would prefer if the dependency on the accelerator-specific random generators were optional.
I brought this up also for CUDA and HIP/ROCm, see #1810 and #1929.

Reply:
Yes, I think we want to outsource RNG in general in the future. For now this shouldn't be optional (to stay consistent with other back-ends) IMO.